service/src/main/java/bio/terra/cbas/dependencies/leonardo/AppUtils.java

```diff
@@ -0,0 +1,118 @@
+package bio.terra.cbas.dependencies.leonardo;
+
+import bio.terra.cbas.common.exceptions.DependencyNotAvailableException;
+import bio.terra.cbas.config.LeonardoServerConfiguration;
+import bio.terra.cbas.config.WdsServerConfiguration;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.broadinstitute.dsde.workbench.client.leonardo.model.AppStatus;
+import org.broadinstitute.dsde.workbench.client.leonardo.model.ListAppResponse;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AppUtils {
+
+  private final LeonardoServerConfiguration leonardoServerConfiguration;
+  private final WdsServerConfiguration wdsServerConfiguration;
+
+  public AppUtils(
+      LeonardoServerConfiguration leonardoServerConfiguration,
+      WdsServerConfiguration wdsServerConfiguration) {
+    this.leonardoServerConfiguration = leonardoServerConfiguration;
+    this.wdsServerConfiguration = wdsServerConfiguration;
+  }
+
+  int appComparisonFunction(ListAppResponse a, ListAppResponse b) {
+    // Name scores: correct name always wins
```

Comment:
It seems [in Terra UI app type along with app name check](https://github.com/DataBiosphere/terra-ui/blob/09fc15b510521e7443c2ab9fda2bf994b78bb27d/src/libs/ajax/data-table-providers/WdsDataTableProvider.ts#L107-L112) takes precedence over other methods to find the WDS proxy url. Maybe we should also include that here so that `WDS` app type in given higher precedence?

Reply:
👍